Cython/CodeWriter.py

```diff
@@ -519,3 +519,276 @@ def visit_CFuncDefNode(self, node):
 
     def visit_StatNode(self, node):
         pass
+
+
+class ExpressionWriter(TreeVisitor):
+
+    def __init__(self, result=None):
+        super(ExpressionWriter, self).__init__()
+        if result is None:
+            result = u""
+        self.result = result
+
+    def write(self, tree):
+        self.visit(tree)
+        return self.result
+
+    def put(self, s):
+        self.result += s
```

Comment:
@scoder IIUC, any python expression has to be written in a single line (unless parentized). I still don't see the point of appending to a list to join at the end.